provisioner/terraform/cleanup_test.go

```diff
@@ -127,25 +129,34 @@ func TestPluginCache_Golden(t *testing.T) {
 	})
 }
 
-func addPluginFile(t *testing.T, fs afero.Fs, pluginPath string, resourcePath string, accessTime time.Time) {
+func addPluginFile(t *testing.T, fs afero.Fs, pluginPath string, resourcePath string, atime, mtime time.Time) {
 	err := fs.MkdirAll(filepath.Join(cachePath, pluginPath), 0o755)
 	require.NoError(t, err, "can't create test folder for plugin file")
 
-	err = fs.Chtimes(filepath.Join(cachePath, pluginPath), accessTime, accessTime)
+	err = fs.Chtimes(filepath.Join(cachePath, pluginPath), atime, mtime)
 	require.NoError(t, err, "can't set times")
 
 	err = afero.WriteFile(fs, filepath.Join(cachePath, pluginPath, resourcePath), []byte("foo"), 0o644)
 	require.NoError(t, err, "can't create test file")
 
-	err = fs.Chtimes(filepath.Join(cachePath, pluginPath, resourcePath), accessTime, accessTime)
+	err = fs.Chtimes(filepath.Join(cachePath, pluginPath, resourcePath), atime, mtime)
 	require.NoError(t, err, "can't set times")
+
+	// as creating a file will update mtime of parent, we also want to
+	// set the mtime of parent to match that of the new child.
+	parent, _ := filepath.Split(filepath.Join(cachePath, pluginPath, resourcePath))
+	parentInfo, err := fs.Stat(parent)
+	require.NoError(t, err, "can't stat parent")
+	if parentInfo.ModTime().After(mtime) {
+		require.NoError(t, fs.Chtimes(parent, atime, mtime), "can't set mtime of parent to match child")
+	}
```

Comment:
review: bit hacky but we want this to look as close to reality as possible

Reply:
just double-check: this should work consistency for different OSes? no risk of flakiness?

Reply:
We use a different tempdir for each test and perform the file operations in serial, so it should probably be OK.